### PR TITLE
fix: fix parallel sqllogictest on stream e2e

### LINKED ISSUE #3051: broken pipe error when running parallel sqllogictest
```
./risedev ci-start ci-3cn-1fe
./risedev slt -p 4566 './e2e_test/streaming/*.slt' -d dev -j 2
```

Will find a lot of broken pipe error, but I have no idea what's the cause.

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

As title, fixed some bugs when update actors. Finally we can support running streaming e2e test. Big thanks for @MrCroxx 's co-debug!!!
```
./risedev` dev ci-3cn-1fe
sqllogictest -p 4566 -d dev  './e2e_test/streaming/**/*.slt' -j 8
```

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)
- Add the 'user-facing changes' label if your PR contains changes that are visible to users (optional)

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
Resolve #3051

### CODE REVIEW

File: src/meta/src/stream/stream_manager.rs [hunk 1]
```diff
@@ -46,7 +46,7 @@ pub struct CreateMaterializedViewContext {
     /// New dispatches to add from upstream actors to downstream actors.
     pub dispatches: HashMap<(ActorId, DispatcherId), Vec<ActorId>>,
     /// Upstream mview actor ids grouped by node id.
-    pub upstream_node_actors: HashMap<WorkerId, Vec<ActorId>>,
+    pub upstream_node_actors: HashMap<WorkerId, HashSet<ActorId>>,
```
Comment (MrCroxx) [inline src/meta/src/stream/stream_manager.rs:49]: I think this is a temporary fix, I'll check why there are duplicates later. 🥰

**Review by MrCroxx — APPROVED** [review]
LGTM!!!!!!!!! Both exhausted, QwQ.

File: src/stream/src/task/mod.rs [hunk 2]
```diff
@@ -147,7 +147,11 @@ impl SharedContext {
 
     #[inline]
     pub fn add_channel_pairs(&self, ids: UpDownActorIds, channels: ConsumableChannelPair) {
-        self.lock_channel_map().insert(ids, channels);
+        assert!(
```
Comment (skyzh) [inline src/stream/src/task/mod.rs:150]: I remembered that there was a none assert before, but I later removed it due to another bug 🤣 Nice to see we've added it back.

**Review by skyzh — APPROVED** [review]
Would you please help also include the CI script change in https://github.com/singularity-data/risingwave/pull/3084, so that parallel e2e test will be running in full cluster mode?

File: src/stream/src/task/mod.rs (hunk 2 above)
Comment (yezizp2012) [reply src/stream/src/task/mod.rs:150]: Okay, I will include the CI script change.

**Pull request merged**